src/domain/headlessCms/persistor.ts

```diff
@@ -0,0 +1,203 @@
+  warn(message: string, ...optionalParams: any) {
+    if (this.logLevel <= LogLevel.WARN) {
+      // eslint-disable-next-line no-console
+      console.warn(message, ...optionalParams);
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  error(message: string, ...optionalParams: any) {
+    if (this.logLevel <= LogLevel.ERROR) {
+      // eslint-disable-next-line no-console
+      console.error(message, ...optionalParams);
+    }
+  }
+}
+
+export interface TimedApolloCachePersistorOptions {
+  /**
+   * The time-to-live for the persisted cache in milliseconds.
+   * Defaults to 10 minutes (600000 milliseconds).
+   */
+  persistedCacheTimeToLive?: number;
+  /**
+   * The logging level to use. Defaults to LogLevel.INFO.
+   */
+  logLevel?: LogLevel;
+}
+
+export class TimedApolloCachePersistor {
+  persistorLocalStorageKey: string;
+  timePersistedLocalStorageKey: string;
+  persistedCacheTimeToLive: number;
+  persistor: CachePersistor<NormalizedCacheObject>;
+  private logger: TimedApolloCachePersistorLogger;
+
+  /**
+   * Creates a new TimedApolloCachePersistor instance to manage the Apollo cache
+   * with an expiration time. This class automatically clears expired caches
+   * and provides methods for persisting and restoring the cache to/from localStorage.
+   *
+   * @param cache The Apollo InMemoryCache instance to be managed.
+   * @param options Options for configuring the persistor.
+   *
+   * @example
+   * const cache = new InMemoryCache();
+   * const persistor = new TimedApolloCachePersistor(cache, {
+   * persistedCacheTimeToLive: 1000 * 60 * 5, // 5 minute expiry
+   * logLevel: LogLevel.DEBUG
+   * });
+   */
+  constructor(
+    cache: InMemoryCache,
+    options: TimedApolloCachePersistorOptions = {}
+  ) {
+    const {
+      persistedCacheTimeToLive = 1000 * 60 * 10, // 10 minutes
+      logLevel = LogLevel.INFO,
+    } = options;
+    this.logger = new TimedApolloCachePersistorLogger(logLevel);
+    this.persistorLocalStorageKey = `kukkuu-ui-cms-${AppConfig.environment}-apollo-cache`;
+    this.timePersistedLocalStorageKey = `kukkuu-ui-cms-${AppConfig.environment}-apollo-cache-persisted-at`;
+    this.persistedCacheTimeToLive = persistedCacheTimeToLive;
+    this.persistor = this.createApolloCachePersistor(cache);
+  }
+
+  /**
+   * @private
+   * Creates and configures the Apollo CachePersistor instance.
+   * This method also handles clearing expired caches from localStorage.
+   * * @param cache The Apollo InMemoryCache instance.
+   * @returns The configured CachePersistor instance.
+   */
+  private createApolloCachePersistor(cache: InMemoryCache) {
+    if (this.hasPersistedCacheExpired()) {
+      try {
+        this.clearLocalStorage();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (e: any) {
+        this.logger.error('Failed to clear local storage:', e.message, e.stack);
+      }
```

Comment:
How about something like this?
```suggestion
      } catch (e: unknown) {
        const errorMessage = 'Failed to clear local storage:';
        if (e instanceof Error) {
          this.logger.error(errorMessage, e.message, e.stack);
        } else {
          this.logger.error(errorMessage, e);
        }
      }
```
To get rid of the eslint disabling, and still handling `Error` type ok, and having a fallback for other raised values.

Reply:
What else can it catch than Errors, which is a base class?

Reply:
Probably not much, I think it's theoretically possible for something else like string, number, boolean, undefined, null, object, array or a function to have been raised. It's ok for me if you leave out the else clause here—up to you.

Reply:
I would say that this evrsion with unknown goes never in else-statement because Error is a base class. Also this is much harder to read and understand. I still think that there is nothing wrong in typing as any in this case.

Reply:
If you want to leave it using the `any` type, go ahead. Just thought I'd try to work with Typescript by using the `unknown` type and type narrowing with the `instanceof`. The else clause is probably not that necessary, mostly defensive fallback for theoretical other types.